code/mob/living/critter/small_animal.dm

```diff
@@ -1855,6 +1855,150 @@ var/list/mob_bird_species = list("smallowl" = /mob/living/critter/small_animal/b
+		HH.can_hold_items = 0
+
+	attackby(obj/item/I, mob/M)
+		if(istype(I, /obj/item/reagent_containers/food/snacks) && ishuman(M) && !isdead(src))
+			src.visible_message("[M] feeds \the [src] some [I].", "[M] feeds you some [I].")
+			for(var/damage_type in src.healthlist)
+				var/datum/healthHolder/hh = src.healthlist[damage_type]
+				hh.HealDamage(5)
+				src.health_brute = min(60, src.health_brute + 6)
+				src.health_burn = min(60, src.health_burn + 6)
+			if(M in src.friends)
+				src.emote("rattle")
+			else
+				if(prob(20))
+					friends += M
+					src.visible_message("[src] hisses happily at the \the [I], and seems a little friendlier with [M].")
+				else
+					src.visible_message("<span class='notice'>[src] hated \the [I] and bit [M]'s hand!</span>")
+					random_brute_damage(M, rand(6,12),1)
+					src.emote("hiss")
+					M.emote("scream")
+			I.Eat(src, src, TRUE)
+			return
+		. = ..()
+
+	attack_hand(mob/M)
+		if ((M.a_intent != INTENT_HARM) && (M in src.friends))
+			if(M.a_intent == INTENT_HELP && src.aggressive)
+				src.visible_message("<span class='notice'>[M] pats [src] on the head in a soothing way. It won't attack anyone now.</span>")
+				src.aggressive = FALSE
+				return
+			else if((M.a_intent == INTENT_DISARM) && !src.aggressive)
+				src.visible_message("<span class='notice'>[M] shakes [src] to awaken it's killer instincts!</span>")
+				src.aggressive = TRUE
+				return
+		..()
+
+	specific_emotes(var/act, var/param = null, var/voluntary = 0)
+		switch (act)
+			if ("scream","hiss")
+				if (src.emote_check(voluntary, 50))
+					playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1, channel=VOLUME_CHANNEL_EMOTE)
+					return "<span class='emote'><b>[src]</b> hisses!</span>"
+			if ("rattle", "snap")
+				if (src.emote_check(voluntary, 50))
+					icon_state = "rattlesnake_rattle"
+					playsound(src, 'sound/musical_instruments/tambourine/tambourine_4.ogg', 80, 1, channel=VOLUME_CHANNEL_EMOTE)
+					SPAWN(1 SECONDS)
+						icon_state = "rattlesnake"
+					return "<span class='emote'><b>[src]</b> rattles it's tail!</span>"
+		return null
+
+	specific_emote_type(var/act)
+		switch (act)
+			if ("scream","hiss", "snip", "snap")
+				return 2
+		return ..()
+
+	critter_attack(var/mob/target)
+		var/datum/targetable/critter/wasp_sting/snake_bite/sting = src.abilityHolder.getAbility(/datum/targetable/critter/wasp_sting/snake_bite)
+
+		if (!sting.disabled && sting.cooldowncheck())
+			sting.handleCast(target)
+			return
+		else
+			return ..()
+
+	seek_target(var/range = 8)
+		. = list()
+		if(!src.aggressive)
+			return .
+		for (var/mob/living/C in hearers(range, src))
+			if (isdead(C)) continue //don't attack the dead
+			if (isintangible(C)) continue //don't attack the AI eye
+			if (istype(C, src.type)) continue //don't attack other snakes
+			if (C in src.friends) continue //don't attack frens :)
+			if (ishuman(C) || issilicon(C))    //creating the snake's defensive behavior
+				switch(GET_DIST(src, C))
```

Comment:
Again, no need for a switch here. Only one statement does anything. Also why does the snake only rattle at humans and silicons, but attack everything?

Reply:
It doesn't attack humans and silicons, it goes after smaller stuff. For instance it won't do much if a person is around, but it will rush to kill a moust if it sees one.